pruning/config/main_config.py

```diff
@@ -5,15 +5,15 @@
 from omegaconf import MISSING
 
 from .datasets import CIFAR10, CIFAR100, BaseDataset, ImageNet1K
+from .iterators import Iterative, Logarithmic, OneShot, PruningIterator
 from .optimizers import SGD, AdamW, BaseOptimizer
 
 
 @dataclass
 class Pruning:
-    step_percent: int = MISSING
-    iterations: int = MISSING
+    iterator: PruningIterator = MISSING
     finetune_epochs: int = MISSING
-    _pruning_checkpoints: tuple[float] = tuple(range(60, 100, 2))
+    _checkpoints_interval: tuple[float, float] = (0.7, 1.0)
```

Comment:
If that is changed to contain start and end we want then it should be changed from tuple to dataclass for example, which contains fields start and end. Because right now it's confusing what it does. Name suggests that these are 2 single checkpoints.

Reply:
Also this restricts ability to save for example on 0.6 and 0.8 only not through the whole interval.

Reply:
Added Interval dataclass to make it more clean to user.